src/dai.js

```diff
@@ -0,0 +1,194 @@
+import DeviceFeature from './device-feature.js';
+import { push, register, deregister } from './dan.js';
+import { RegistrationError, ArgumentError } from './exceptions.js';
+
+export default class {
+    constructor(option) {
+        this.api_url = option['api_url'];
+        this.device_model = option['device_model'];
+        this.device_addr = option['device_addr'];
+        this.device_name = option['device_name'];
+        this.persistent_binding = option['persistent_binding'] || false;
+        this.username = option['username'];
+        this.extra_setup_webpage = option['extra_setup_webpage'] || '';
+        this.device_webpage = option['device_webpage'] || '';
+
+        this.register_callback = option['register_callback'];
+        this.on_register = option['on_register'];
+        this.on_deregister = option['on_deregister'];
+        this.on_connect = option['on_connect'];
+        this.on_disconnect = option['on_disconnect'];
+
+        this.push_interval = typeof option['push_interval'] != 'undefined' ? option['push_interval'] : 1;
+        this.interval = option['interval'] || {};
+
+        this.device_features = {};
+        this.flags = {};
+
+        this.on_signal = this.on_signal.bind(this);
+        this.on_data = this.on_data.bind(this);
+
+        this.parse_df_profile(option, 'idf');
+        this.parse_df_profile(option, 'odf');
+    }
+
+    push_data(df_name) {
+        if (this.device_features[df_name].push_data == null)
+            return;
+        let _df_interval = this.interval[df_name] || this.push_interval;
+        console.debug(`${df_name} : ${this.flags[df_name]} [message / ${_df_interval} s]`);
+        let _push_interval = setInterval(() => {
+            let _data = this.device_features[df_name].push_data();
+            console.log(_data);
+            if (!this.flags[df_name] && typeof _data != 'undefined') {
+                clearInterval(_push_interval);
+            }
+            else {
+                push(df_name, _data);
+            }
+        }, _df_interval * 1000);
+    }
+
+    on_signal(signal, df_list) {
+        console.log(`Receive signal: ${signal}, ${df_list}`);
+        if ('CONNECT' == signal) {
+            df_list.forEach(df_name => {
+                if (!this.flags[df_name]) {
+                    this.flags[df_name] = true;
+                    this.push_data(df_name);
+                }
```

Comment:
在 function 中用 Arrow anti pattern
```suggestion
                if (this.flags[df_name]) {
                    return;
                }
                this.flags[df_name] = true;
                this.push_data(df_name);
```